packetbeat/protos/mysql/mysql.go

```diff
@@ -311,8 +357,25 @@ func mysqlMessageParser(s *mysqlStream) (bool, bool) {
 			m.size = uint64(m.end - m.start)
 			logp.Debug("mysqldetailed", "Message complete. remaining=%d",
 				len(s.data[s.parseOffset:]))
-			return true, true
 
+			// PREPARE_OK packet for Prepared Statement
+			// a trick for classify special OK packet
+			if m.isOK && m.packetLength == 12 {
+				idPtr := s.data[m.start+5 : m.start+13]
+				m.statementID = int(idPtr[0]) | int(idPtr[1])<<8 | int(idPtr[2])<<16 | int(idPtr[3])<<24
+				m.numberOfFields = int(idPtr[4]) | int(idPtr[5])<<8
+				m.numberOfParameter = int(idPtr[6]) | int(idPtr[7])<<8
```

Comment:
one can use `binary.BigEndian.UintX` here.

Reply:
it's little-endian format